### PR TITLE
Add execute stage

### PR DESCRIPTION
**Pull request by dbarashev**

### CODE REVIEW

**Comment by dbarashev** [comment]
@edgarzhavoronkov Output.kt was moved from DockerStage with some changes in FileOutput

File: build.gradle [hunk 1]
```diff
@@ -131,6 +138,7 @@ publishing {
         mavenMidasProto(MavenPublication) {
```
Comment (edgarzhavoronkov) [inline build.gradle]: probably Midas here is not relevant

File: src/main/kotlin/com/bardsoftware/papeeria/backend/BaseServer.kt [hunk 2]
```diff
@@ -175,6 +176,10 @@ fun start(arg: BaseServerArgs, service: BackendService, serverName: String) = ma
     server
   }
   service.pubsub?.let {
+    if (arg.sub == null) {
+      System.err.println("Missing subscription name.")
+      exitProcess(1)
+    }
     LOG.info("Listening to PubSub subscription ${arg.sub!!}")
```
Comment (edgarzhavoronkov) [inline src/main/kotlin/com/bardsoftware/papeeria/backend/BaseServer.kt]: Looks like checking for nullity here is redundant, if you are doing it above

File: src/main/kotlin/com/bardsoftware/papeeria/backend/ExecuteStage.kt [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        }
+        val copyStderr = async {
+          proc.errorStream.copyTo(out.stderrPipe)
+        }
+        proc.waitFor(1, TimeUnit.MINUTES)
+        val exitCode = proc.exitValue()
+        copyStdout.await()
+        copyStderr.await()
+
+        LOG.debug("Exit code =$exitCode")
```
Comment (edgarzhavoronkov) [inline src/main/kotlin/com/bardsoftware/papeeria/backend/ExecuteStage.kt]: nit: whitespace after `=`

File: src/main/kotlin/com/bardsoftware/papeeria/backend/Output.kt [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        while (scanner.hasNextLine()) {
+          println(scanner.nextLine())
+        }
+      }
+    }
+  }
+}
+
+class FileOutput(private val fileOut: PrintWriter) : BaseOutput() {
+  private val counter = AtomicInteger(2)
```
Comment (edgarzhavoronkov) [inline src/main/kotlin/com/bardsoftware/papeeria/backend/Output.kt]: Probably this numbers deserves some comments

Comment (dbarashev) [reply src/main/kotlin/com/bardsoftware/papeeria/backend/Output.kt]: done

File: src/main/kotlin/com/bardsoftware/papeeria/backend/ExecuteStage.kt (hunk 3 above)
Comment (dbarashev) [reply src/main/kotlin/com/bardsoftware/papeeria/backend/ExecuteStage.kt]: done

File: src/main/kotlin/com/bardsoftware/papeeria/backend/BaseServer.kt (hunk 2 above)
Comment (dbarashev) [reply src/main/kotlin/com/bardsoftware/papeeria/backend/BaseServer.kt]: refactored a little bit

File: build.gradle (hunk 1 above)
Comment (dbarashev) [reply build.gradle]: yuo, thanks

**Comment by dbarashev** [comment]
@edgarzhavoronkov now with some tests

**Review by edgarzhavoronkov — APPROVED** [review]
LGTM

**Pull request merged**